percona/pmm/pmm.go

```diff
@@ -224,15 +224,23 @@ func sidecarContainerV2(pgc *v2.PerconaPGCluster) corev1.Container {
 			},
 			{
 				Name:  "PMM_AGENT_PRERUN_SCRIPT",
-				Value: agentPrerunScript(pgc.Spec.PMM.QuerySource),
+				Value: agentPrerunScript(pgc.Spec.PMM.QuerySource, pgc),
+			},
+			{
+				Name:  "PMM_AGENT_PATHS_TEMPDIR",
+				Value: "/tmp",
 			},
 		},
 	}
 
-	if pgc.CompareVersion("2.3.0") >= 0 {
+	if pgc.CompareVersion("2.7.0") >= 0 {
```

Comment:
On the V2 unit test that we already have, we should change the cr version from 2.6 to 2.7 and assert properly the new fields. If we want to do more comprehensive and complete checks we can add a test case for <2.6 and one that works for >=2.7